controllers/manilaapi_controller.go

```diff
@@ -638,30 +638,43 @@ func (r *ManilaAPIReconciler) reconcileNormal(ctx context.Context, instance *man
 	//
 	// check for required OpenStack secret holding passwords for service/admin user and add hash to the vars map
 	//
-	ctrlResult, err := r.getSecret(ctx, helper, instance, instance.Spec.TransportURLSecret, &configVars)
+
+	hash, ctrlResult, err := ensureSecret(
```

Comment:
Is it necessary to ensureSecret()? I think you should be able to use getConfigSecret().

The reason I say this is the top manila controller doesn't even create the individual deployments until after the TransportURLSecret exists.

Reply:
Considering that we don't have to check a particular entry within the `transportURL` secret, I think we can safely use `getConfigSecret()` here. You're right that in theory we're stopped by the top-level controller if transportURL is not ready, and it will create a reconciliation loop.
I'll update this part (same thing applies to scheduler and share) in my next update.

Reply:
Done